### PR TITLE
Redesign + TypeScript + Styled Components + NextJS + MUI components + rewrite

### LINKED ISSUE #34: Download button non-functional on mobile
**Describe the bug**
On an Android and a iPhone when pressing DOWNLOAD button
on the main page nothing happens. But it does work on my
Desktop computer using Chrome and if I enable "Desktop site" on
my Android phone it also works as expected. So this is mobile
specific.

**What kind of a bug is it?**
User interface on the site is not working as expected.

**To Reproduce**
0. On a mobile device
1. Go to https://dahliaos.io
2. Press Download, no animation
3. Release DOWNLOAD, nothing happens

**Expected behavior**
Because DOWNLOAD is a button it should animate when pressed and when
released it should "goto" the linked page or section. The "LEARN MORE"
button directly above it does animate and on release renders the expected
information. The DOWNLOAD button does nothing.

**Screenshots**
N/A

**Browser (please complete the following information):**
 - Desktop: Chrome
 - Mobile: Chrome (on Android), Safari (on iPhone)

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by Looskie**
## Description

This PR includes the entire rewrite of the website, introducing React/NextJS, TypeScript, official MUI components, a full redesign, and a kick-ass intro animation.

Fixes #34 

## Type of change

Please tick the relevant option by putting an X inside the bracket

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) document

Closes #38 in favor for this one.

This will require some changes to our documentation, as its going to remove any styles that the documentation depends on.

API Routes
`/api` (Welcome page)
`/api/latest_version` (returns latest version from githubs api)

### CODE REVIEW

File: hooks/useGithubReleases.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import useSWR from "swr";
+import { Releases } from "../types/types";
+
+export const useGithubReleases = () => {
+  const { data, error } = useSWR(
```
Comment (Looskie) [inline hooks/useGithubReleases.ts:5]: Use explicit typing here

File: pages/api/hello.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
```
Comment (Looskie) [inline pages/api/hello.ts:1]: lets just remove this entire thing, unless we need a serverless api route for something

**Review by Looskie — COMMENTED** [review]
lgtm!

**Review by allansrc — APPROVED** [review]
0a6d4500-ab2e-11ea-867d-28d76998f760

**Comment by boon-cpu** [comment]
LGTM :+1:

**Review by quintenvandamme — APPROVED** [review]
LGTM

**Comment by alii** [comment]
software programmer engineering computer bill gate

**Review by alii — APPROVED** [review]

**Review by bleonard252 — CHANGES_REQUESTED** [review]
Why is `public/favicon.ico` the Vercel icon? Shouldn't it be the dahliaOS logo? `public/images/favicon/favicon.ico` is correct so maybe `public/favicon.ico` is vestigial or should match (for situations where the HTML isn't checked)?

**Comment by Looskie** [comment]
> Why is `public/favicon.ico` the Vercel icon? Shouldn't it be the dahliaOS logo? `public/images/favicon/favicon.ico` is correct so maybe `public/favicon.ico` is vestigial or should match (for situations where the HTML isn't checked)?

yeah i just forgot to remove the vercel logo, its never used in our codebase but comes with the create-next-app template

**Review by bleonard252 — APPROVED** [review]
LGTM 👍

**Review by pxseu — APPROVED** [review]

**Review by larsb24 — APPROVED** [review]
LGTM!

**Pull request merged**